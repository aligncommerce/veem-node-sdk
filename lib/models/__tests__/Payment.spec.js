import PaymentModel from 'models/Payment'

describe('Payment Model', () => {
  describe('request', () => {
    it('should match the snapshot', () => {
      expect(PaymentModel.request.schema).toMatchSnapshot()
    })

    it('should define a valid schema', () => {
      expect(PaymentModel.request.schema).to.be.validJsonSchema
    })
  })

  describe('response', () => {
    it('should match the snapshot', () => {
      expect(PaymentModel.response.schema).toMatchSnapshot()
    })

    it('should define a valid schema', () => {
      expect(PaymentModel.response.schema).to.be.validJsonSchema
    })
  })
})
