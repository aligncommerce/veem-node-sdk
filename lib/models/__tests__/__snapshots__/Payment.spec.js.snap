// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Model request should match the snapshot 1`] = `
Object {
  "properties": Object {
    "amount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "approveAutomatically": Object {
      "type": "boolean",
    },
    "attachments": Object {
      "properties": Object {
        "name": Object {
          "type": "string",
        },
        "referenceId": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "ccEmails": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "clientId": Object {
      "type": "string",
    },
    "dueDate": Object {
      "type": "string",
    },
    "exchangeRate": Object {
      "properties": Object {
        "fromAmount": Object {
          "type": "string",
        },
        "fromCurrency": Object {
          "type": "string",
        },
        "recipientAccountEmail": Object {
          "type": "string",
        },
        "toAmount": Object {
          "type": "string",
        },
        "toCountry": Object {
          "type": "string",
        },
        "toCurrency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "exchangeRateQuoteId": Object {
      "type": "string",
    },
    "externalInvoiceRefId": Object {
      "type": "string",
    },
    "notes": Object {
      "type": "string",
    },
    "payee": Object {
      "properties": Object {
        "businessName": Object {
          "type": "string",
        },
        "countryCode": Object {
          "type": "string",
        },
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
        "phone": Object {
          "type": "string",
        },
        "type": Object {
          "enum": Array [
            "Incomplete",
            "Business",
            "Personal",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
    "payeeAmount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "paymentAction": Object {
      "type": "string",
    },
    "paymentApprovalRequest": Object {
      "accountId": Object {
        "type": "number",
      },
      "userId": Object {
        "type": "number",
      },
    },
    "purposeOfPayment": Object {
      "type": "string",
    },
    "status": Object {
      "type": "string",
    },
  },
  "title": "PaymentRequest",
  "type": "object",
}
`;

exports[`Payment Model response should match the snapshot 1`] = `
Object {
  "properties": Object {
    "amount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "attachments": Object {
      "properties": Object {},
      "type": "object",
    },
    "batchItemId": Object {
      "type": "number",
    },
    "ccEmails": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "claimLink": Object {
      "type": "string",
    },
    "clientId": Object {
      "type": "string",
    },
    "dueDate": Object {
      "type": "string",
    },
    "exchangeRate": Object {
      "properties": Object {
        "expiry": Object {
          "type": "number",
        },
        "fromAmount": Object {
          "type": "number",
        },
        "fromCurrency": Object {
          "type": "string",
        },
        "id": Object {
          "type": "string",
        },
        "rate": Object {
          "optional": true,
          "type": "number",
        },
        "timeCreated": Object {
          "optional": true,
          "type": "string",
        },
        "toAmount": Object {
          "type": "number",
        },
        "toCurrency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "exchangeRateQuoteId": Object {
      "type": "string",
    },
    "externalInvoiceRefId": Object {
      "type": "string",
    },
    "id": Object {
      "type": "number",
    },
    "notes": Object {
      "type": "string",
    },
    "payee": Object {
      "properties": Object {
        "businessName": Object {
          "type": "string",
        },
        "countryCode": Object {
          "type": "string",
        },
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
        "phone": Object {
          "type": "string",
        },
        "type": Object {
          "enum": Array [
            "Incomplete",
            "Business",
            "Personal",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
    "payeeAmount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "paymentAction": Object {
      "type": "string",
    },
    "paymentApproval": Object {
      "properties": Object {
        "approvalStatus": Object {
          "type": "string",
        },
        "approverNumber": Object {
          "type": "number",
        },
        "approverNumberRequired": Object {
          "type": "number",
        },
        "userApprovalList": Object {
          "items": Object {
            "properties": Object {
              "approvalStatus": Object {
                "type": "string",
              },
              "email": Object {
                "type": "string",
              },
              "firstName": Object {
                "type": "string",
              },
              "lastName": Object {
                "type": "string",
              },
              "middleName": Object {
                "type": "string",
              },
            },
            "type": "object",
          },
          "type": "array",
        },
      },
      "type": "object",
    },
    "paymentApprovalRequest": Object {
      "accountId": Object {
        "type": "number",
      },
      "userId": Object {
        "type": "number",
      },
    },
    "purposeOfPayment": Object {
      "type": "string",
    },
    "pushPaymentInfo": Object {
      "amount": Object {
        "properties": Object {
          "amount": Object {
            "type": "number",
          },
          "currency": Object {
            "type": "string",
          },
        },
        "type": "obect",
      },
      "pushPaymentInfo": Object {
        "reference": "string",
        "type": "string",
      },
    },
    "requestId": Object {
      "type": "string",
    },
    "status": Object {
      "type": "string",
    },
    "timeCreated": Object {
      "type": "string",
    },
    "timeUpdated": Object {
      "type": "string",
    },
  },
  "title": "PaymentResponse",
  "type": "object",
}
`;
